Mathlib/Topology/ExtremallyDisconnected.lean

```diff
@@ -139,8 +134,163 @@ protected theorem CompactT2.Projective.extremallyDisconnected [CompactSpace X] [
     exact hx.1
 #align compact_t2.projective.extremally_disconnected CompactT2.Projective.extremallyDisconnected
 
+end
+
+section
+
+variable {A B C D E : Type u} [TopologicalSpace A] [TopologicalSpace B] [TopologicalSpace C]
+  [TopologicalSpace D] [TopologicalSpace E]
+
+/-- Lemma 2.4 in [Gleason, *Projective topological spaces*][gleason1958]:
+a continuous surjection $\pi$ from a compact space $D$ to a Fréchet space $A$ restricts to
+a compact subset $E$ of $D$, such that $\pi$ maps $E$ onto $A$ and satisfies the
+"Zorn subset condition", where $\pi(E_0) \ne A$ for any proper closed subset $E_0 \subsetneq E$.
+
+Proof. Apply Zorn's lemma on the closed subsets $E$ of $D$ such that $\pi(E) = A$. -/
+lemma exists_compact_surjective_zorn_subset [T1Space A] [CompactSpace D] {π : D → A}
+    (π_cont : Continuous π) (π_surj : π.Surjective) : ∃ E : Set D, CompactSpace E ∧ π '' E = univ ∧
+    ∀ E₀ : Set E, E₀ ≠ univ → IsClosed E₀ → E.restrict π '' E₀ ≠ univ := by
```

Comment:
The statement of this lemma isn't very idiomatic in my opinion (because it's viewing a `E : Set D` as a type and then using it as a topological space). That's not to say this is *necessarily* the wrong way to go about it, because sometimes it is just the right thing to do. But I wonder whether this is nicer?
```lean
lemma exists_compact_surjective_zorn_subset [T1Space A] [CompactSpace D] {π : D → A}
    (π_cont : Continuous π) (π_surj : π.Surjective) : ∃ E : Type _, CompactSpace E ∧ ∃ ι : E → D, Embedding ι ∧ (π ∘ ι).Surjective ∧
    ∀ E₀ : Set E, E₀ ≠ univ → IsClosed E₀ → ¬ (π ∘ ι ∘ ((↑) : E₀ → E)).Surjective 
```
It's totally possible it's not, I'm just asking you to consider it, and if you feel it's worse, then don't implement it and just leave a comment here.

Reply:
The statement of your lemma doesn't seem to compile for me: it says `E` needs to be a topological space before we can say it's compact, and we don't know this without having it live as a subset of `D`.